server/controller/user/inviteUser.ts

```diff
@@ -0,0 +1,50 @@
+import { BaseProtocol, DB, Schemas } from '@server/db'
+import { RoleNames, User, UserInvitation, UserStatus } from '@core/meta/user'
+import { ActivityLogRepository, UserRepository, UserInvitationRepository } from '@server/repository'
+import { Assessment, ActivityLogMessage } from '@core/meta/assessment'
+
+export const inviteUser = async (
+  props: {
+    assessment: Assessment
+    countryIso: string
+    cycleUuid: string
+    email: string
+    roleName: RoleNames
+    user: User
+  },
+  client: BaseProtocol = DB
+): Promise<{ userInvitation: UserInvitation; user: User }> => {
+  const { assessment, email, user } = props
+  const schemaName = Schemas.getName(assessment)
+
+  return client.tx(async (t) => {
+    let userToInvite = await UserRepository.read({ user: { email } }, client)
+    if (!userToInvite) {
+      userToInvite = await UserRepository.create({ user: { email, name: '' } })
+      userToInvite.status = UserStatus.invitationPending
```

Comment:
by default, when creating a user from the repository , his status is already invitationPending. if not, please add it to the default column